AnkiDroid/src/main/java/com/ichi2/widget/DeckPickerWidgetConfig.kt

```diff
@@ -0,0 +1,197 @@
+
+            finish()
+        }
+
+        findViewById<FloatingActionButton>(R.id.fabWidgetDeckPicker).setOnClickListener {
+            showDeckSelectionDialog()
+        }
+
+        loadSelectedDecksFromPreferences()
+        updateViewVisibility()
+    }
+
+    private fun showDeckSelectionDialog() {
+        lifecycleScope.launch {
+            val decks = fetchDecks()
+            displayDeckSelectionDialog(decks)
+        }
+    }
+
+    private suspend fun fetchDecks(): List<SelectableDeck> {
+        return withContext(Dispatchers.IO) {
+            SelectableDeck.fromCollection(includeFiltered = false)
+        }
+    }
+
+    private fun displayDeckSelectionDialog(decks: List<SelectableDeck>) {
+        val dialog = DeckSelectionDialog.newInstance(
+            title = getString(R.string.select_deck_title),
+            summaryMessage = null,
+            keepRestoreDefaultButton = false,
+            decks = decks
+        )
+        dialog.show(supportFragmentManager, "DeckSelectionDialog")
+    }
+
+    override fun onDeckSelected(deck: SelectableDeck?) {
+        if (deck != null) {
+            if (deckAdapter.itemCount >= 5) {
+                showSnackbar(getString(R.string.deck_limit_reached))
+            } else {
+                deckAdapter.addDeck(deck)
+                updateViewVisibility()
+            }
+        }
+    }
+
+    private fun updateViewVisibility() {
+        val noDecksPlaceholder = findViewById<View>(R.id.no_decks_placeholder)
+        val widgetConfigContainer = findViewById<View>(R.id.widgetConfigContainer)
+
+        if (deckAdapter.itemCount > 0) {
+            noDecksPlaceholder.visibility = View.GONE
+            widgetConfigContainer.visibility = View.VISIBLE
+        } else {
+            noDecksPlaceholder.visibility = View.VISIBLE
+            widgetConfigContainer.visibility = View.GONE
+        }
+    }
+
+    private val itemTouchHelperCallback = object : ItemTouchHelper.SimpleCallback(
+        ItemTouchHelper.UP or ItemTouchHelper.DOWN,
+        0
+    ) {
+        override fun onMove(
+            recyclerView: RecyclerView,
+            viewHolder: RecyclerView.ViewHolder,
+            target: RecyclerView.ViewHolder
+        ): Boolean {
+            val fromPosition = viewHolder.bindingAdapterPosition
+            val toPosition = target.bindingAdapterPosition
+            deckAdapter.moveDeck(fromPosition, toPosition)
+            return true
+        }
+
+        override fun onSwiped(viewHolder: RecyclerView.ViewHolder, direction: Int) {
+            // No swipe action
+        }
+    }
+
+    private fun saveSelectedDecksToPreferences() {
+        val sharedPreferences = getSharedPreferences("DeckPickerWidgetPrefs", Context.MODE_PRIVATE)
+        val editor = sharedPreferences.edit()
+        val selectedDecks = deckAdapter.decks.map { it.deckId.toString() }
+        editor.putString("selected_decks_$appWidgetId", selectedDecks.joinToString(","))
+        editor.apply()
```

Comment:
non-functional refactor. You'll need to import `edit {`

```suggestion
        val sharedPreferences = getSharedPreferences("DeckPickerWidgetPrefs", Context.MODE_PRIVATE)
        val selectedDecks = deckAdapter.decks.map { it.deckId.toString() }
        getSharedPreferences("DeckPickerWidgetPrefs", Context.MODE_PRIVATE).edit {
            putString("selected_decks_$appWidgetId", selectedDecks.joinToString(","))
        }
```